hatchery/hatchery.go

```diff
@@ -401,6 +403,27 @@ func launch(w http.ResponseWriter, r *http.Request) {
 		Config.Logger.Printf("Debug: Nextflow is not enabled: skipping Nextflow resources creation")
 	}
 
+	if strings.Contains(strings.ToLower(Config.ContainersMap[hash].Name), "gen3-licensed") {
```

Comment:
here and in `getWorkspaceFlavor()` you check the flavor of the container implicitly based on the name, i think it would be cleaner to add a flag to the container config so we can explicitly configure the "gen3-licensed" flavor.

In general code shouldn't depend on display names, for example we might want to use more user-friendly flavor names in production. I know that in `getWorkspaceFlavor()` there's existing code that checks the name and you probably based this code on that, but "jupyter" is a bit more generic and i think we can do better here

Reply:
I second this comment, we should and another config field at container level to indicate whether this container will be using this license distribution feature, instead of relying on container names

Reply:
The license section of the container config now has a `workspace-flavor` field.